File: src/lessons/recursion/cons/LastEntity.java
package lessons.recursion.cons;

import static plm.core.ValueSerializer.*;

import lessons.recursion.cons.universe.ConsEntity;
import lessons.recursion.cons.universe.RecList;

public class LastEntity extends ConsEntity {

  public void run()
  {
    int count = getTestCount();
    for (int i = 0; i < count; i++) {
      Object[] param = (Object[])deserialize(getTest(i));
      setTestResult(i, serialize(last(RecList.fromArray((int[]) param[0]))));
    }
  }

  /* BEGIN TEMPLATE */
  int last(RecList seq)
  {
    /* BEGIN SOLUTION */
    if (seq.tail == null)
      return seq.head;
    return last(seq.tail);
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}

File: src/lessons/recursion/cons/LastEntity.py
def run():
    count = getTestCount()
    for i in range(count):
        param = deserialize(getTest(i))
        setTestResult(i, serialize(last(toRecListIfArray(param[0]))))

# BEGIN TEMPLATE
def last(list):
# BEGIN SOLUTION
  if list.tail == None:
    return list.head
  return last(list.tail)
# END SOLUTION
# END TEMPLATE
